test/BumpMapping_test.cpp

```diff
@@ -0,0 +1,262 @@
+    cv::split(cvImage, channels);
+    channels[1] = cvDepth;
+    cv::merge(channels, cvImage);
+    cv::cvtColor(cvImage, cvImage, cv::COLOR_RGB2BGR);
+    cv::flip(cvImage, cvImage, 0);
+
+    return cvImage.clone();
+}
+
+// simple polar plot
+void plotSonarTest(cv::Mat3f image, double maxRange, double maxAngleX) {
+    cv::Mat1b imagePlot = cv::Mat1b::zeros(500, 500);
+    cv::Point2f centerPlot(imagePlot.cols / 2, 0);
+    double factor = imagePlot.rows / maxRange;
+    double slope = 2 * maxAngleX * (1.0 / (image.cols - 1));
+
+    for (int j = 0; j < image.rows; ++j) {
+        for (int i = 0; i < image.cols; ++i) {
+            double distance = image[j][i][1] * maxRange;
+            double alpha = slope * i - maxAngleX;
+
+            cv::Point2f tempPoint(distance * sin(alpha), distance * cos(alpha));
+            tempPoint = tempPoint * factor + centerPlot;
+            imagePlot[(uint) tempPoint.y][(uint) tempPoint.x] = 255 * image[j][i][0];
+        }
+    }
+
+    cv::Mat3b imagePlotMap;
+    cv::applyColorMap(imagePlot, imagePlotMap, cv::COLORMAP_HOT);
+
+    cv::line( imagePlotMap, centerPlot, cv::Point2f(maxRange * sin(maxAngleX) * factor,
+              maxRange * cos(maxAngleX) * factor) + centerPlot, cv::Scalar(255), 1, CV_AA);
+
+    cv::line( imagePlotMap, centerPlot, cv::Point2f(maxRange * sin(-maxAngleX) * factor,
+              maxRange * cos(maxAngleX) * factor) + centerPlot, cv::Scalar(255), 1, CV_AA);
+
+    cv::imshow("Normal Depth Map", image);
+    cv::imshow("Sonar Plot Test", imagePlotMap);
+    cv::waitKey();
+}
+
+BOOST_AUTO_TEST_CASE(differentNormalMaps_TestCase) {
+    float maxRange = 20.0f;
+    float fovX = M_PI / 3;  // 60 degrees
+    float fovY = M_PI / 3;  // 60 degrees
+
+    osg::ref_ptr<osg::Group> simpleRoot = createSimpleScene();
+    osg::ref_ptr<osg::Group> bumpRoot = createBumpMapSimpleScene();
+
+    cv::Mat cvSimple = computeNormalDepthMap(simpleRoot, maxRange, fovX, fovY);
+    cv::Mat cvBump = computeNormalDepthMap(bumpRoot, maxRange, fovX, fovY);
+
+    std::vector<cv::Mat> simpleChannels, bumpChannels;
+    cv::split(cvSimple, simpleChannels);
+    cv::split(cvBump, bumpChannels);
+
+    // assert that the normal matrixes are different
+    BOOST_CHECK(are_equals(simpleChannels[0], bumpChannels[0]) == false);
+
+    // assert that the depth matrixes are equals
+    BOOST_CHECK(are_equals(simpleChannels[1], bumpChannels[1]) == true);
+
+    // plot sonar sample output
+    plotSonarTest(cvSimple, maxRange, fovX * 0.5);
+    plotSonarTest(cvBump, maxRange, fovX * 0.5);
+}
+
+BOOST_AUTO_TEST_CASE(multiTextureScene_TestCase) {
+    float maxRange = 25.0f;
+    float fovX = M_PI / 4;  // 45 degrees
+    float fovY = M_PI / 4;  // 45 degrees
+
+    osg::ref_ptr<osg::Group> bumpRoot = createBumpMapMultiScene();
+    cv::Mat cvBump = computeNormalDepthMap(bumpRoot, maxRange, fovX, fovY);
+
+    // plot sonar sample output
+    plotSonarTest(cvBump, maxRange, fovX * 0.5);
```

Comment:
I am missing some numerical validation. You could take some pixel reference and compare the values.